lib/utils.js

```diff
@@ -0,0 +1,191 @@
+  }
+
+  var host_port;
+  if (hostList.length > 1) {
+    result.database.host = [];
+    result.database.port = [];
+    for (i = 0; i < hostList.length; i++) {
+      host_port = hostList[i].split(":");
+      assert.ok(host_port[0] && host_port[0].length > 0);
+      result.database.host.push(host_port[0]);
+      if (host_port.length > 1) {
+        result.database.port.push(host_port[1]);
+      } else {
+        result.database.port.push('27017');
+      }
+    }
+  }
+  else {
+    host_port = hostList[0].split(":");
+    assert.ok(host_port[0] && host_port[0].length > 0);
+    result.database.host = host_port[0];
+    if (host_port.length > 1) {
+      result.database.port = host_port[1];
+    } else {
+      result.database.port = '27017';
+    }
+  }
+
+
+  //Verify the config to be returned
+  assert.ok(result.database.name.length > 0);
+  assert.ok(result.database.auth.user.length > 0);
+  assert.ok(result.database.auth.pass.length > 0);
+
+
+  return result;
+};
+
+/**
+ * This will return a connection URL like:
+ * host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+ *
+ * With the intention that it can be prepended with "mongodb://[username]:[password]@"
+ * and appended with "/[database][?options]
+ *
+ * @param host This is expected in the format: "host1" or "host1,host2" or ["host1", "host2"]
+ * @param port This is expected in the format: 123 or "123" or "123,456" or ["123", "456"] or [123, 456]
+ *
+ * @returns {string} host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+ */
+exports.parseConnectionUrl = function (host, port){
+  var hosts = [];
+  var ports = [];
+
+  //normalise host to an array
+  if(! host) {
+    hosts  = ['localhost'];
+  } else if('string' === typeof host && host.split(',').length <= 1){
+    hosts = [host];
+  } else if ('string' === typeof host && host.split(',').length > 1){
+    hosts = host.split(',');
+  } else if (Array.isArray(host)){
+    hosts = host;
+  }
+
+  //normalise port to an array
+  if(! port){
+    ports = [27017];
+  } else if('string' === typeof port && port.split(',').length <= 1 || 'number' === typeof port){
+    ports = [port];
+  } else if ('string' === typeof port && port.split(',').length > 1){
+    ports = port.split(',');
+  } else if (Array.isArray(port)){
+    ports = port;
+  }
+
+  var connectionHosts = [];
+
+  for( var hostIndex=0; hostIndex < hosts.length; hostIndex++ ){
+    connectionHosts[hostIndex] = hosts[hostIndex];
+
+    //if only one port specified, use it on all hosts, otherwise assume number of ports and hosts are equal
+    if(ports.length === 1){
+      connectionHosts[hostIndex] += ":" + ports[0];
+    } else {
+      connectionHosts[hostIndex] += ":" + ports[hostIndex];
+    }
+  }
+
+  return connectionHosts.join(',');
+}
+
+
+/**
+ * Returns the driver options URL encoded for the connection string; e.g.: w=1&j=true
+ *
+ * @param options an Array of options
+ *
+ * @returns {string}
+ */
+exports.parseConnectionUrlOptions = function (options) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)